net: l2: ethernet: bring in TX-Injection mode to l2 ethernet

VivekUppunda opened the pull request:
This set of changes brings in raw packet tx injection mode feature to ethernet and removes it from being a Wi-Fi only feature.

It was earlier envisaged as Wi-Fi net management feature only. It also removes promiscuous mode setting in Wi-Fi mode setting.

VivekUppunda committed:
net: l2: ethernet: bring in TX-Injection mode to l2 ethernet

This set of changes brings in raw packet tx injection mode
feature to ethernet and removes it from being a Wi-Fi only feature.

It was earlier envisaged as Wi-Fi net management feature only.

Signed-off-by: Vivekananda Uppunda <vivekananda.uppunda@nordicsemi.no>

jukkar reviewed (commented):
Overall looks good, but could you split the PR to two commits, first one introducing the tx-injection mode, and the other one removing the promiscuous mode from wifi shell, this way we do not mix commits that change different things.

VivekUppunda committed:
net: wifi_shell: Remove TX-Injection and Promiscuous mode from Wi-Fi mode

Remove TX-Injection and Promiscuous mode setting from Wi-Fi mode shell
command. These commands are being moved to ethernet l2 layer

Signed-off-by: Vivekananda Uppunda <vivekananda.uppunda@nordicsemi.no>

VivekUppunda commented:
> Overall looks good, but could you split the PR to two commits, first one introducing the tx-injection mode, and the other one removing the promiscuous mode from wifi shell, this way we do not mix commits that change different things.

Moved the tx-injection and promiscuous mode removal from wifi-shell to a new commit. 
Thank you
